### PR TITLE
Jp 1222 FGS image2 regression test

### PR DESCRIPTION
**Pull request by ddavis-stsci**
Add the nightly test covering the FGS image 2 tests to the regtest suite.

Closes JP-1222 & #4421

### CODE REVIEW

**Comment by jdavies-st** [comment]
It looks like FGS_IMAGE data do not get resampled in our pipeline.  Should they?  They certainly have `assign_wcs` run on them and have distortion reffiles.  One could add that exptype to

https://github.com/spacetelescope/jwst/blob/481d8ea3f0499e0132a5ffea01e28d1ef514ce0a/jwst/pipeline/calwebb_image2.py#L43

to give it a try.  Currently, because they are not listed there, resampling (and the resulting `_i2d.fits` file) does not happen.

Are there any other EXP_TYPES that need to go through this pipeline and get resampled?  FGS_FOCUS also has a distortion model applied in `assign_wcs`.

**Comment by ddavis-stsci** [comment]
Adding 'FGS_IMAGE' to image_exptypes in calwebb_image.py and rerunning the pipeline results in an error 

> strun --steps.flat_field.save_results=True calwebb_image2.cfg jw86500007001_02101_00001_GUIDER2_rate.fits

...
>2020-01-13 14:20:20,531 - stpipe.Image2Pipeline.resample - INFO - No NIRSpec DIRZPARS reffile
2020-01-13 14:20:20,533 - stpipe.Image2Pipeline.resample - INFO -   setting: pixfrac=1.0
2020-01-13 14:20:20,533 - stpipe.Image2Pipeline.resample - INFO -   setting: kernel='square'
2020-01-13 14:20:20,533 - stpipe.Image2Pipeline.resample - INFO -   setting: fillval='INDEF'
2020-01-13 14:20:20,533 - stpipe.Image2Pipeline.resample - INFO -   setting: weight_type='exptime'
ERROR RUNNING STEP 'Image2Pipeline':
    Invalid output frame shape: (0, 0)
Traceback (most recent call last):

**Comment by ddavis-stsci** [comment]
Adding 'FGS_IMAGE' to image_exptypes in calwebb_image.py and rerunning the pipeline results in an error 

> strun --steps.flat_field.save_results=True calwebb_image2.cfg jw86500007001_02101_00001_GUIDER2_rate.fits

...
>2020-01-13 14:20:20,531 - stpipe.Image2Pipeline.resample - INFO - No NIRSpec DIRZPARS reffile
2020-01-13 14:20:20,533 - stpipe.Image2Pipeline.resample - INFO -   setting: pixfrac=1.0
2020-01-13 14:20:20,533 - stpipe.Image2Pipeline.resample - INFO -   setting: kernel='square'
2020-01-13 14:20:20,533 - stpipe.Image2Pipeline.resample - INFO -   setting: fillval='INDEF'
2020-01-13 14:20:20,533 - stpipe.Image2Pipeline.resample - INFO -   setting: weight_type='exptime'
ERROR RUNNING STEP 'Image2Pipeline':
    Invalid output frame shape: (0, 0)
Traceback (most recent call last):

**Comment by ddavis-stsci** [comment]
The rate file header has been updated with values from a current SDP run and FGS_IMAGE has been added to the exposure types to be processed in calwebb_image2.py for the resample step. 
Local pytest results,
= 3 passed, 253 deselected, 3 warnings in 126.89s (0:02:06) =

File: CHANGES.rst [hunk 1]
```diff
@@ -31,6 +31,8 @@ master_background
 pipeline
 --------
 
+- Added FGS_IMAGE to the exposure types to be processed in calwebb_image2.py [#4421]
```
Comment (hbushouse) [inline CHANGES.rst]: FGS_IMAGE exposures are already processed in calwebb_image2. What's actually changed is that the ``resample`` step is now applied to them. So "... exposure types to apply resampling in calwebb_image2.py".

**Pull request merged**